src/main/java/org/jboss/staxmapper/XMLMapperImpl.java

```diff
@@ -100,10 +108,24 @@ public void deparseDocument(XMLContentWriter contentWriter, XMLStreamWriter stre
         contentWriter.writeContent(new FormattingXMLStreamWriter(streamWriter));
     }
 
-    @SuppressWarnings({ "unchecked" })
+    private XMLElementReader getParser(final QName name) {
```

Comment:
we are casting anyway later on when getting parser, so doing it in this method vs at caller is the same.
Cast is done from <?> to <T> and that was also before.